test/jabbax/parser_test.exs

```diff
@@ -0,0 +1,82 @@
+defmodule Jabbax.ParserTest do
+  use ExUnit.Case
+  use Plug.Test
+  use Jabbax.Document
+
+  def parse(conn, parsers) do
+    Plug.Parsers.call(conn, Plug.Parsers.init(parsers: parsers))
+  end
+
+  test "JSON API document" do
+    connection = conn(:post, "/", Poison.encode!(%{
+      "data" => %{
+        "id" => "1",
+        "type" => "user",
+        "attributes" => %{
+          "name" => "Sample User"
+        }
+      },
+      "jsonapi" => %{
+        "version" => "1.0"
+      }
+    }))
+    |> put_req_header("content-type", "application/vnd.api+json")
+    |> parse([Jabbax.Parser])
+
+    assert connection.body_params == %Document{
+      data: %Resource{
+        id: "1",
+        type: "user",
+        attributes: %{
+          "name" => "Sample User"
+        }
+      }
+    }
+  end
+
+  test "misc content type" do
+    connection = conn(:post, "/", Poison.encode!(%{
+      "data" => %{
+        "id" => "1",
+        "type" => "user",
+        "attributes" => %{
+          "name" => "Sample User"
+        }
+      },
+      "jsonapi" => %{
+        "version" => "1.0"
+      }
+    }))
+    |> parse([Jabbax.Parser])
+
+    assert connection.body_params == %{}
+  end
+
+  test "no jsonapi key" do
```

Comment:
:+1: Thanks. 

(Actually I've moved this test from deserializer_test as it was written to use conn, but in reality it's deserializer's concern, not parser's so I've moved it back and fixed the test bug that you've caught.)